plugins/woocommerce/src/Admin/Features/Blueprint/RestApi.php

```diff
@@ -0,0 +1,202 @@
+				),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/export',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'export' ),
+					'permission_callback' => array( $this, 'check_permission' ),
+					'args'                => array(
+						'steps'         => array(
+							'description'       => __( 'A list of plugins to install', 'woocommerce' ),
+							'type'              => 'array',
+							'items'             => 'string',
+							'default'           => array(),
+							'sanitize_callback' => function ( $value ) {
+								return array_map(
+									function ( $value ) {
+										return sanitize_text_field( $value );
+									},
+									$value
+								);
+							},
+							'required'          => false,
+						),
+						'export_as_zip' => array(
+							'description' => __( 'Export as a zip file', 'woocommerce' ),
+							'type'        => 'boolean',
+							'default'     => false,
+							'required'    => false,
+						),
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Check if the current user has permission to perform the request.
+	 *
+	 * @return bool|\WP_Error
+	 */
+	public function check_permission() {
+		if ( ! wc_rest_check_manager_permissions( 'settings', 'read' ) ) {
+			return new \WP_Error( 'woocommerce_rest_cannot_view', __( 'Sorry, you cannot list resources.', 'woocommerce' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Handle the export request.
+	 *
+	 * @param \WP_REST_Request $request The request object.
+	 * @return \WP_HTTP_Response The response object.
+	 */
+	public function export( $request ) {
+		$steps         = $request->get_param( 'steps' );
+		$export_as_zip = $request->get_param( 'export_as_zip' );
+		$exporter      = new ExportSchema();
+
+		$data = $exporter->export( $steps, $export_as_zip );
+
+		if ( $export_as_zip ) {
+			$zip  = new ZipExportedSchema( $data );
+			$data = $zip->zip();
+			$data = site_url( str_replace( ABSPATH, '', $data ) );
+		}
+
+		return new \WP_HTTP_Response(
+			array(
+				'data' => $data,
+				'type' => $export_as_zip ? 'zip' : 'json',
+			)
+		);
+	}
+
+	/**
+	 * Handle the import request.
+	 *
+	 * @return \WP_HTTP_Response The response object.
+	 * @throws \InvalidArgumentException If the import fails.
+	 */
+	public function import() {
+
+		// Check for nonce to prevent CSRF.
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+		if ( ! isset( $_POST['blueprint_upload_nonce'] ) || ! \wp_verify_nonce( $_POST['blueprint_upload_nonce'], 'blueprint_upload_nonce' ) ) {
+			return new \WP_HTTP_Response(
+				array(
+					'status'  => 'error',
+					'message' => __( 'Invalid nonce', 'woocommerce' ),
+				),
+				400
+			);
+		}
+
+		// phpcs:ignore
+		if ( ! empty( $_FILES['file'] ) && $_FILES['file']['error'] === UPLOAD_ERR_OK ) {
```

Comment:
nit: I'm not familiar with file uploads in PHP 😅  but saw that we use `is_uploaded_file` and `is_file` functions to check if the file is uploaded in upload_theme() API. Maybe we can use the same functions to check if the file is uploaded here to make it more robust. 

https://github.com/woocommerce/woocommerce/blob/b2f4828c16a583d7fb35f353d6ba8d08c8da1824/plugins/woocommerce/src/Admin/API/Themes.php#L76-L80

Reply:
That's a good suggestion. I think checking `$_FILES['file']['error'] === UPLOAD_ERR_OK` does almost the same thing already.  Let's come back to this later. We need to refactor the upload validation part anyway.